pkg/metricfamily/sort.go

```diff
@@ -0,0 +1,131 @@
+package metricfamily
+
+import (
+	"sort"
+
+	clientmodel "github.com/prometheus/client_model/go"
+)
+
+func SortMetrics(family *clientmodel.MetricFamily) (bool, error) {
+	sort.Sort(MetricsByTimestamp(family.Metric))
+	return true, nil
+}
+
+type MetricsByTimestamp []*clientmodel.Metric
```

Comment:
Do we really need to use these "old" interfaces? I think you can simply use the `sort.Slice` func with the `func (m MetricsByTimestamp) Less(i int, j int) bool` as parameter. 😊

Reply:
Agreed :+1: This is just a first mechanical refactor (hopefully) without actual logic changes.

Reply:
right! logic and opinionated clean up will follow shortly!